### DISCUSSION TITLE
Unable to run LAMMPS using trained model

### DISCUSSION THREAD

**Discussion by yzjin** [deepmodeling/deepmd-kit#1025] [Q&A]
I have installed the deepmd-kit using conda, and successfully trained a model locally. However, when I tried to use LAMMPS coming with deepmd-kit to run simulation using the dumped model, it leads to the following error:
`ERROR: Unrecognized pair style 'deepmd' (src/force.cpp:285)`
`Last command: pair_style        deepmd graph.pb`

The `lmp -h` command works totally fine, but deepmd is not an acceptable pair style under the documentation provided by the command. Has anyone encountered such error before? Thanks in advance.

**Comment by njzjz** [comment]
`lmp -h` also provides the package information. Could you post the full output here? Besides, are you sure the `lmp` you executed was downloaded from our channel?

**Reply by yzjin** [reply]
Thank you for your prompt reply. I downloaded the lmp via `conda install deepmd-kit=*=*gpu lammps-dp=*=*gpu -c deepmodeling` as recommended by the documentation. Running `which lmp` gives me `../miniconda3/envs/deepmd/bin/lmp` so I'm pretty sure it is downloaded using conda. Here's the output for `lmp -h`:


Large-scale Atomic/Molecular Massively Parallel Simulator - 29 Oct 2020

Usage example: lmp -var t 300 -echo screen -in in.alloy

List of command line options supported by this LAMMPS executable:

-echo none/screen/log/both  : echoing of input script (-e)
-help                       : print this help message (-h)
-in filename                : read input from file, not stdin (-i)
-kokkos on/off ...          : turn KOKKOS mode on or off (-k)
-log none/filename          : where to send log output (-l)
-mpicolor color             : which exe in a multi-exe mpirun cmd (-m)
-nocite                     : disable writing log.cite file (-nc)
-package style ...          : invoke package command (-pk)
-partition size1 size2 ...  : assign partition sizes (-p)
-plog basename              : basename for partition logs (-pl)
-pscreen basename           : basename for partition screens (-ps)
-restart2data rfile dfile ... : convert restart to data file (-r2data)
-restart2dump rfile dgroup dstyle dfile ... 
                            : convert restart to dump file (-r2dump)
-reorder topology-specs     : processor reordering (-r)
-screen none/filename       : where to send screen output (-sc)
-suffix gpu/intel/opt/omp   : style suffix to apply (-sf)
-var varname value          : set index style variable (-v)

OS: Linux 5.4.0-77-generic on x86_64

Compiler: GNU C++ 7.3.0 with OpenMP 4.5
C++ standard: C++11
MPI v3.1: MPICH Version:	3.3.2
MPICH Release date:	Tue Nov 12 21:23:16 CST 2019
MPICH ABI:	13:8:1

Active compile time flags:

-DLAMMPS_GZIP
-DLAMMPS_PNG
-DLAMMPS_JPEG
-DLAMMPS_SMALLBIG
sizeof(smallint): 32-bit
sizeof(imageint): 32-bit
sizeof(tagint):   32-bit
sizeof(bigint):   64-bit

Installed packages:

ASPHERE BODY CLASS2 COLLOID CORESHELL DIPOLE GRANULAR KSPACE MANYBODY MC MISC 
MOLECULE OPT PERI REPLICA RIGID SHOCK SNAP SRD USER-COLVARS USER-DEEPMD 
USER-FEP USER-MEAMC USER-MISC USER-PHONON USER-PLUMED USER-REAXC 

List of individual style options included in this LAMMPS executable

* Atom styles:

angle           atomic          body            bond            charge          
dipole          ellipsoid       full            hybrid          line            
molecular       peri            sphere          template        tri             

* Integrate styles:

respa           verlet          verlet/split    

* Minimize styles:

cg              fire            fire/old        hftn            quickmin        
sd              

* Pair styles:

adp             agni            airebo          airebo/morse    atm             
beck            body/nparticle  body/rounded/polygon            
body/rounded/polyhedron         bop             born            born/coul/dsf   
born/coul/dsf/cs                born/coul/long  born/coul/long/cs               
born/coul/msm   born/coul/wolf  born/coul/wolf/cs               brownian        
brownian/poly   buck            buck/coul/cut   buck/coul/long  
buck/coul/long/cs               buck/coul/msm   buck/long/coul/long             
buck/mdf        colloid         comb            comb3           cosine/squared  
coul/cut        coul/cut/soft   coul/debye      coul/diel       coul/dsf        
coul/long       coul/long/cs    coul/long/soft  coul/msm        coul/shield     
coul/slater/cut coul/slater/long                coul/streitz    coul/wolf       
coul/wolf/cs    reax            dpd             dpd/tstat       drip            
dsmc            e3b             eam             eam/alloy       eam/alloy/opt   
eam/cd          eam/cd/old      eam/fs          eam/fs/opt      eam/opt         
edip            edip/multi      eim             extep           gauss           
gauss/cut       gayberne        gran/hertz/history              gran/hooke      
gran/hooke/history              granular        gw              gw/zbl          
hbond/dreiding/lj               hbond/dreiding/morse            hybrid          
hybrid/overlay  ilp/graphene/hbn                kolmogorov/crespi/full          
kolmogorov/crespi/z             lcbop           lebedeva/z      lennard/mdf     
line/lj         list            lj96/cut        lj/charmm/coul/charmm           
lj/charmm/coul/charmm/implicit  lj/charmm/coul/long             
lj/charmm/coul/long/opt         lj/charmm/coul/long/soft        
lj/charmm/coul/msm              lj/charmmfsw/coul/charmmfsh     
lj/charmmfsw/coul/long          lj/class2       lj/class2/coul/cut              
lj/class2/coul/cut/soft         lj/class2/coul/long             
lj/class2/coul/long/cs          lj/class2/coul/long/soft        lj/class2/soft  
lj/cubic        lj/cut          lj/cut/coul/cut lj/cut/coul/cut/soft            
lj/cut/coul/debye               lj/cut/coul/dsf lj/cut/coul/long                
lj/cut/coul/long/cs             lj/cut/coul/long/opt            
lj/cut/coul/long/soft           lj/cut/coul/msm lj/cut/coul/wolf                
lj/cut/dipole/cut               lj/cut/dipole/long              lj/cut/opt      
lj/cut/soft     lj/cut/tip4p/cut                lj/cut/tip4p/long               
lj/cut/tip4p/long/opt           lj/cut/tip4p/long/soft          lj/expand       
lj/expand/coul/long             lj/gromacs      lj/gromacs/coul/gromacs         
lj/long/coul/long               lj/long/coul/long/opt           
lj/long/dipole/long             lj/long/tip4p/long              lj/mdf          
lj/sf/dipole/sf lj/smooth       lj/smooth/linear                lj/sf           
local/density   lubricate       lubricateU      lubricateU/poly lubricate/poly  
meam/spline     meam/sw/spline  meam/c          meam            mie/cut         
momb            morse           morse/opt       morse/smooth/linear             
morse/soft      nb3b/harmonic   nm/cut          nm/cut/coul/cut nm/cut/coul/long                
deepmd          peri/eps        peri/lps        peri/pmb        peri/ves        
polymorphic     reax/c          rebo            resquared       snap            
soft            srp             sw              table           tersoff         
tersoff/mod     tersoff/mod/c   tersoff/table   tersoff/zbl     tip4p/cut       
tip4p/long      tip4p/long/soft tri/lj          ufm             ufm/opt         
vashishta       vashishta/table yukawa          yukawa/colloid  zbl             
zero            

* Bond styles:

class2          fene            fene/expand     gromos          harmonic        
harmonic/shift  harmonic/shift/cut              hybrid          morse           
nonlinear       quartic         special         table           zero            

* Angle styles:

charmm          class2          cosine          cosine/delta    cosine/periodic 
cosine/shift    cosine/shift/exp                cosine/squared  dipole          
fourier         fourier/simple  harmonic        hybrid          quartic         
table           zero            

* Dihedral styles:

charmm          charmmfsw       class2          cosine/shift/exp                
fourier         harmonic        helix           hybrid          multi/harmonic  
nharmonic       opls            quadratic       spherical       table           
table/cut       zero            

* Improper styles:

class2          cossq           cvff            distance        fourier         
harmonic        hybrid          ring            umbrella        zero            

* KSpace styles:

ewald           ewald/dipole    ewald/dipole/spin               ewald/disp      
msm             msm/cg          pppm            pppm/cg         pppm/dipole     
pppm/dipole/spin                pppm/disp       pppm/disp/tip4p pppm/dplr       
pppm/stagger    pppm/tip4p      

* Fix styles

accelerate/cos  adapt           adapt/fep       addforce        addtorque       
append/atoms    atom/swap       ave/atom        ave/chunk       ave/correlate   
ave/correlate/long              ave/histo       ave/histo/weight                
ave/time        aveforce        balance         bond/break      bond/create     
bond/create/angle               bond/swap       box/relax       cmap            
colvars         controller      deform          deposit         ave/spatial     
ave/spatial/sphere              dplr            drag            dt/reset        
efield          ehex            electron/stopping               enforce2d       
evaporate       external        ffl             filter/corotate flow/gauss      
freeze          gcmc            gld             gle             gravity         
grem            halt            heat            hyper/global    hyper/local     
imd             indent          ipi             langevin        lineforce       
momentum        momentum/chunk  move            msst            neb             
nph             nph/asphere     nph/body        nph/sphere      nphug           
npt             npt/asphere     npt/body        npt/cauchy      npt/sphere      
numdiff         nve             nve/asphere     nve/asphere/noforce             
nve/body        nve/limit       nve/line        nve/noforce     nve/sphere      
nve/tri         nvk             nvt             nvt/asphere     nvt/body        
nvt/sllod       nvt/sphere      oneway          orient/bcc      orient/eco      
orient/fcc      pafi            phonon          pimd            planeforce      
plumed          pour            press/berendsen print           propel/self     
property/atom   qeq/comb        qeq/reax        rattle          reax/c/bonds    
reax/c/species  recenter        restrain        rhok            rigid           
rigid/nph       rigid/nph/small rigid/npt       rigid/npt/small rigid/nve       
rigid/nve/small rigid/nvt       rigid/nvt/small rigid/small     setforce        
shake           smd             spring          spring/chunk    spring/rg       
spring/self     srd             store/force     store/state     temp/berendsen  
temp/csld       temp/csvr       temp/rescale    tfmc            
thermal/conductivity            ti/spring       tmd             ttm             
ttm/mod         tune/kspace     vector          viscosity       viscous         
wall/body/polygon               wall/body/polyhedron            wall/colloid    
wall/ees        wall/gran       wall/gran/region                wall/harmonic   
wall/lj1043     wall/lj126      wall/lj93       wall/morse      wall/piston     
wall/reflect    wall/reflect/stochastic         wall/region     wall/region/ees 
wall/srd        widom           

* Compute styles:

ackland/atom    adf             aggregate/atom  angle           angle/local     
angmom/chunk    basal/atom      body/local      bond            bond/local      
centro/atom     centroid/stress/atom            chunk/atom      
chunk/spread/atom               cluster/atom    cna/atom        cnp/atom        
com             com/chunk       contact/atom    coord/atom      damage/atom     
dihedral        dihedral/local  dilatation/atom dipole/chunk    displace/atom   
entropy/atom    erotate/asphere erotate/rigid   erotate/sphere  
erotate/sphere/atom             event/displace  fep             fragment/atom   
global/atom     group/group     gyration        gyration/chunk  gyration/shape  
gyration/shape/chunk            heat/flux       hexorder/atom   hma             
improper        improper/local  inertia/chunk   ke              ke/atom         
ke/rigid        momentum        msd             msd/chunk       msd/nongauss    
omega/chunk     orientorder/atom                pair            pair/local      
pe              pe/atom         plasticity/atom pressure        
pressure/cylinder               property/atom   property/chunk  property/local  
rdf             reduce          reduce/chunk    reduce/region   rigid/local     
slice           sna/atom        snad/atom       snap            snav/atom       
stress/atom     stress/mop      stress/mop/profile              temp            
temp/asphere    temp/body       temp/chunk      temp/com        temp/cs         
temp/deform     temp/partial    temp/profile    temp/ramp       temp/region     
temp/rotate     temp/sphere     ti              torque/chunk    vacf            
vcm/chunk       viscosity/cos   

* Region styles:

block           cone            cylinder        intersect       plane           
prism           sphere          union           

* Dump styles:

atom            cfg             custom          dcd             image           
local           movie           xtc             xyz             

* Command styles

balance         change_box      create_atoms    create_bonds    create_box      
delete_atoms    delete_bonds    reset_ids       displace_atoms  dynamical_matrix                
group2ndx       hyper           info            minimize        ndx2group       
neb             prd             read_data       read_dump       read_restart    
replicate       rerun           reset_atom_ids  reset_mol_ids   run             
set             tad             temper          temper/grem     temper/npt      
third_order     velocity        write_coeff     write_data      write_dump      
write_restart

**Reply by njzjz** [reply]
I can find `deepmd` pair style here:

> Pair styles:
> 
> adp agni airebo airebo/morse atm
> beck body/nparticle body/rounded/polygon
> body/rounded/polyhedron bop born born/coul/dsf
> born/coul/dsf/cs born/coul/long born/coul/long/cs
> born/coul/msm born/coul/wolf born/coul/wolf/cs brownian
> brownian/poly buck buck/coul/cut buck/coul/long
> buck/coul/long/cs buck/coul/msm buck/long/coul/long
> buck/mdf colloid comb comb3 cosine/squared
> coul/cut coul/cut/soft coul/debye coul/diel coul/dsf
> coul/long coul/long/cs coul/long/soft coul/msm coul/shield
> coul/slater/cut coul/slater/long coul/streitz coul/wolf
> coul/wolf/cs reax dpd dpd/tstat drip
> dsmc e3b eam eam/alloy eam/alloy/opt
> eam/cd eam/cd/old eam/fs eam/fs/opt eam/opt
> edip edip/multi eim extep gauss
> gauss/cut gayberne gran/hertz/history gran/hooke
> gran/hooke/history granular gw gw/zbl
> hbond/dreiding/lj hbond/dreiding/morse hybrid
> hybrid/overlay ilp/graphene/hbn kolmogorov/crespi/full
> kolmogorov/crespi/z lcbop lebedeva/z lennard/mdf
> line/lj list lj96/cut lj/charmm/coul/charmm
> lj/charmm/coul/charmm/implicit lj/charmm/coul/long
> lj/charmm/coul/long/opt lj/charmm/coul/long/soft
> lj/charmm/coul/msm lj/charmmfsw/coul/charmmfsh
> lj/charmmfsw/coul/long lj/class2 lj/class2/coul/cut
> lj/class2/coul/cut/soft lj/class2/coul/long
> lj/class2/coul/long/cs lj/class2/coul/long/soft lj/class2/soft
> lj/cubic lj/cut lj/cut/coul/cut lj/cut/coul/cut/soft
> lj/cut/coul/debye lj/cut/coul/dsf lj/cut/coul/long
> lj/cut/coul/long/cs lj/cut/coul/long/opt
> lj/cut/coul/long/soft lj/cut/coul/msm lj/cut/coul/wolf
> lj/cut/dipole/cut lj/cut/dipole/long lj/cut/opt
> lj/cut/soft lj/cut/tip4p/cut lj/cut/tip4p/long
> lj/cut/tip4p/long/opt lj/cut/tip4p/long/soft lj/expand
> lj/expand/coul/long lj/gromacs lj/gromacs/coul/gromacs
> lj/long/coul/long lj/long/coul/long/opt
> lj/long/dipole/long lj/long/tip4p/long lj/mdf
> lj/sf/dipole/sf lj/smooth lj/smooth/linear lj/sf
> local/density lubricate lubricateU lubricateU/poly lubricate/poly
> meam/spline meam/sw/spline meam/c meam mie/cut
> momb morse morse/opt morse/smooth/linear
> morse/soft nb3b/harmonic nm/cut nm/cut/coul/cut nm/cut/coul/long
> **deepmd** peri/eps peri/lps peri/pmb peri/ves
> polymorphic reax/c rebo resquared snap
> soft srp sw table tersoff
> tersoff/mod tersoff/mod/c tersoff/table tersoff/zbl tip4p/cut
> tip4p/long tip4p/long/soft tri/lj ufm ufm/opt
> vashishta vashishta/table yukawa yukawa/colloid zbl
> zero

**Reply by yzjin** [reply]
Hmmm that's weird. I guess the problem is that I also installed a non deepmd-compliant lammps via conda and this lammps has messed up some of the files behind. I'm thinking about reinstalling the deepmd-kit and lammps-dp to fix this issue, but I'm not sure whether my model, potentially trained on an earlier version of the deepmd-kit, could be used if I reinstall both packages. Do you know whether the old model and the newer version of lmp are compatible with each other?

**Comment by xuekuiwin** [comment]
hello, I have a similar problem: ERROR: Unrecognized pair style 'deepmd' (../force.cpp:285), Have you solved it yet? Can you share it with me?

**Comment by JTaozhang** [comment]
I meet the same problem, Have you found some solution?

**Reply by CloudZhang-6** [reply]
I met the same problem this morning. reinstalling the latest version of deepmd solved my problem:
conda create -n deepmd deepmd-kit=*=*gpu libdeepmd=*=*gpu lammps-dp cudatoolkit=11.3 horovod -c https://conda.deepmodeling.org

**Comment by michaelmacisaac** [comment]
I have encountered a similar issue. I find that if I activate the environment in my terminal or my batch script, that the batch job then works

**Comment by caolz** [comment]
I have a same problem.

LAMMPS (2 Aug 2023)
OMP_NUM_THREADS environment is not set. Defaulting to 1 thread. (src/comm.cpp:98)
  using 1 OpenMP thread(s) per MPI task
Open of file /home/ustb-clz/software/intel/oneapi/intelpython/latest/envs/deepmd/lib/deepmd_lmp/dpplugin.so failed: /home/ustb-clz/software/intel/oneapi/intelpython/latest/envs/deepmd/lib/deepmd_lmp/dpplugin.so: undefined symbol: MPI_Bcast
Loaded 0 plugins from /home/ustb-clz/software/intel/oneapi/intelpython/latest/envs/deepmd/lib/deepmd_lmp

The 'box' command has been removed and will be ignored

Reading data file ...
  triclinic box = (0 0 0) to (23.504033 20.35509 19.190963) with tilt (11.752017 11.752017 6.7850299)
WARNING: Triclinic box skew is large. LAMMPS will run inefficiently. (src/domain.cpp:220)
  1 by 1 by 1 MPI processor grid
  reading atoms ...
  243 atoms
  read_data CPU = 0.001 seconds
ERROR: Unrecognized pair style 'deepmd' (src/force.cpp:271)
Last command: pair_style     deepmd graph.pb

**Comment by Gemstrom** [comment]
Hello, I've also encountered the same issue, I wonder if anyone have an idea on how to solve it. Thanks in advance.
Error message:
ERROR: Unrecognized pair style 'deepmd' (src/force.cpp:271)
Last command: pair_style	deepmd graph.pb
More info:
My deepmd version is 2.2.5 and is installed via conda-forge command on this page of document:https://docs.deepmodeling.com/projects/deepmd/en/master/getting-started/install.html#install-with-conda, and I'm running via bash command. 
By running $lmp_serial -h command, no deepmd is found in the pair-style section. Upgrading deepmd does not solve this issue.

**Reply by njzjz** [reply]
The plugin is only built for the MPI version. The serial one does not build the shared library so no plugin can be used.

**Reply by yjia5** [reply]
I used deepmdkit 2.2.5. The MPI version does not include "deepmd" pair style, either.

Large-scale Atomic/Molecular Massively Parallel Simulator - 2 Aug 2023

Usage example: lmp_mpi -var t 300 -echo screen -in in.alloy

List of command line options supported by this LAMMPS executable:

-echo none/screen/log/both  : echoing of input script (-e)
-help                       : print this help message (-h)
-in none/filename           : read input from file or stdin (default) (-i)
-kokkos on/off ...          : turn KOKKOS mode on or off (-k)
-log none/filename          : where to send log output (-l)
-mdi '<mdi flags>'          : pass flags to the MolSSI Driver Interface
-mpicolor color             : which exe in a multi-exe mpirun cmd (-m)
-cite                       : select citation reminder style (-c)
-nocite                     : disable citation reminder (-nc)
-nonbuf                     : disable screen/logfile buffering (-nb)
-package style ...          : invoke package command (-pk)
-partition size1 size2 ...  : assign partition sizes (-p)
-plog basename              : basename for partition logs (-pl)
-pscreen basename           : basename for partition screens (-ps)
-restart2data rfile dfile ... : convert restart to data file (-r2data)
-restart2dump rfile dgroup dstyle dfile ... 
                            : convert restart to dump file (-r2dump)
-reorder topology-specs     : processor reordering (-r)
-screen none/filename       : where to send screen output (-sc)
-skiprun                    : skip loops in run and minimize (-sr)
-suffix gpu/intel/opt/omp   : style suffix to apply (-sf)
-var varname value          : set index style variable (-v)

OS: Linux "AlmaLinux 8.8 (Sapphire Caracal)" 4.18.0-477.21.1.el8_8.x86_64 x86_64

Compiler: GNU C++ 12.3.0 with OpenMP 4.5
C++ standard: C++11
MPI v4.0: MPICH Version:      4.1.2
MPICH Release date: Wed Jun  7 15:22:45 CDT 2023
MPICH ABI:          15:1:3

Accelerator configuration:


Active compile time flags:

-DLAMMPS_GZIP
-DLAMMPS_PNG
-DLAMMPS_JPEG
-DLAMMPS_EXCEPTIONS
-DLAMMPS_SMALLBIG
sizeof(smallint): 32-bit
sizeof(imageint): 32-bit
sizeof(tagint):   32-bit
sizeof(bigint):   64-bit

Available compression formats:

Extension: .gz     Command: gzip
Extension: .bz2    Command: bzip2
Extension: .zst    Command: zstd
Extension: .xz     Command: xz
Extension: .lzma   Command: xz
Extension: .lz4    Command: lz4


Installed packages:

ASPHERE BODY BROWNIAN CLASS2 COLLOID COLVARS CORESHELL DIPOLE EXTRA-COMPUTE 
EXTRA-DUMP EXTRA-FIX EXTRA-MOLECULE EXTRA-PAIR GRANULAR H5MD KIM KSPACE 
MANYBODY MC MEAM MISC ML-HDNNP ML-IAP ML-PACE ML-QUIP ML-SNAP MOLECULE MPIIO 
NETCDF OPT PERI PHONON PLUGIN PLUMED PYTHON REAXFF REPLICA RIGID SHOCK SRD 
USER-MLIP VORONOI 

List of individual style options included in this LAMMPS executable

* Atom styles:

angle           atomic          body            bond            charge          
dipole          ellipsoid       full            hybrid          line            
molecular       peri            sphere          template        tri             

* Integrate styles:

respa           verlet          verlet/split    

* Minimize styles:

cg              fire/old        fire            hftn            quickmin        
sd              

* Pair styles:

mlip            adp             agni            airebo          airebo/morse    
atm             beck            body/nparticle  body/rounded/polygon            
body/rounded/polyhedron         bop             born            born/coul/dsf   
born/coul/dsf/cs                born/coul/long  born/coul/long/cs               
born/coul/msm   born/coul/wolf  born/coul/wolf/cs               born/gauss      
brownian        brownian/poly   buck            buck/coul/cut   buck/coul/long  
buck/coul/long/cs               buck/coul/msm   buck/long/coul/long             
buck/mdf        colloid         comb            comb3           cosine/squared  
coul/cut        coul/cut/global coul/debye      coul/diel       coul/dsf        
coul/exclude    coul/long       coul/long/cs    coul/msm        coul/slater/cut 
coul/slater/long                coul/streitz    coul/wolf       coul/wolf/cs    
reax            mesont/tpm      dsmc            e3b             eam             
eam/alloy       eam/alloy/opt   eam/cd          eam/cd/old      eam/fs          
eam/fs/opt      eam/he          eam/opt         edip            edip/multi      
eim             extep           gauss           gauss/cut       gayberne        
gran/hertz/history              gran/hooke      gran/hooke/history              
granular        gw              gw/zbl          harmonic/cut    
hbond/dreiding/lj               hbond/dreiding/morse            hdnnp           
hybrid          hybrid/overlay  hybrid/scaled   kim             lcbop           
lennard/mdf     line/lj         list            lj96/cut        
lj/charmm/coul/charmm           lj/charmm/coul/charmm/implicit  
lj/charmm/coul/long             lj/charmm/coul/long/opt         
lj/charmm/coul/msm              lj/charmmfsw/coul/charmmfsh     
lj/charmmfsw/coul/long          lj/class2       lj/class2/coul/cut              
lj/class2/coul/long             lj/class2/coul/long/cs          lj/cubic        
lj/cut          lj/cut/coul/cut lj/cut/coul/debye               lj/cut/coul/dsf 
lj/cut/coul/long                lj/cut/coul/long/cs             
lj/cut/coul/long/opt            lj/cut/coul/msm lj/cut/coul/wolf                
lj/cut/dipole/cut               lj/cut/dipole/long              lj/cut/opt      
lj/cut/sphere   lj/cut/tip4p/cut                lj/cut/tip4p/long               
lj/cut/tip4p/long/opt           lj/expand       lj/expand/coul/long             
lj/expand/sphere                lj/gromacs      lj/gromacs/coul/gromacs         
lj/long/coul/long               lj/long/coul/long/opt           
lj/long/dipole/long             lj/long/tip4p/long              lj/mdf          
lj/relres       lj/sf/dipole/sf lj/smooth       lj/smooth/linear                
lj/sf           local/density   lubricate       lubricateU      lubricateU/poly 
lubricate/poly  meam            meam/c          meam/ms         meam/spline     
meam/sw/spline  mie/cut         mliap           momb            morse           
morse/opt       morse/smooth/linear             nb3b/harmonic   nm/cut          
nm/cut/coul/cut nm/cut/coul/long                nm/cut/split    pace            
pace/extrapolation              peri/eps        peri/lps        peri/pmb        
peri/ves        polymorphic     python          quip            reaxff          
reax/c          rebo            resquared       snap            soft            
srp             srp/react       sw              sw/angle/table  sw/mod          
table           tersoff         tersoff/mod     tersoff/mod/c   tersoff/table   
tersoff/zbl     threebody/table tip4p/cut       tip4p/long      tracker         
tri/lj          ufm             ufm/opt         vashishta       vashishta/table 
wf/cut          ylz             yukawa          yukawa/colloid  zbl             
zero            

* Bond styles:

class2          fene            fene/expand     fene/nm         gaussian        
gromos          harmonic        harmonic/restrain               harmonic/shift  
harmonic/shift/cut              hybrid          morse           nonlinear       
quartic         special         table           zero            

* Angle styles:

charmm          class2          cosine          cosine/delta    cosine/periodic 
cosine/shift    cosine/shift/exp                cosine/squared  dipole          
fourier         fourier/simple  gaussian        harmonic        hybrid          
quartic         table           zero            

* Dihedral styles:

charmm          charmmfsw       class2          cosine/shift/exp                
fourier         harmonic        helix           hybrid          multi/harmonic  
nharmonic       opls            quadratic       spherical       table           
table/cut       zero            

* Improper styles:

class2          cossq           cvff            distance        fourier         
harmonic        hybrid          ring            umbrella        zero            

* KSpace styles:

ewald           ewald/dipole    ewald/dipole/spin               ewald/disp      
ewald/disp/dipole               msm             msm/cg          pppm            
pppm/cg         pppm/dipole     pppm/dipole/spin                pppm/disp       
pppm/disp/tip4p pppm/stagger    pppm/tip4p      

* Fix styles

accelerate/cos  acks2/reax      acks2/reaxff    adapt           addforce        
addtorque       alchemy         append/atoms    atom/swap       ave/atom        
ave/chunk       ave/correlate   ave/correlate/long              ave/grid        
ave/histo       ave/histo/weight                ave/time        aveforce        
balance         bond/break      bond/create     bond/create/angle               
bond/swap       box/relax       brownian        brownian/asphere                
brownian/sphere charge/regulation               cmap            colvars         
controller      damping/cundall deform          deposit         ave/spatial     
ave/spatial/sphere              lb/pc           lb/rigid/pc/sphere              
client/md       drag            dt/reset        efield          efield/tip4p    
ehex            electron/stopping               electron/stopping/fit           
enforce2d       evaporate       external        ffl             filter/corotate 
flow/gauss      freeze          gcmc            gld             gle             
gravity         grem            halt            heat            heat/flow       
hyper/global    hyper/local     imd             indent          ipi             
langevin        lineforce       mol/swap        momentum        momentum/chunk  
move            msst            neb             nph             nph/asphere     
nph/body        nph/sphere      nphug           npt             npt/asphere     
npt/body        npt/cauchy      npt/sphere      numdiff         numdiff/virial  
nve             nve/asphere     nve/asphere/noforce             nve/body        
nve/limit       nve/line        nve/noforce     nve/sphere      nve/tri         
nvk             nvt             nvt/asphere     nvt/body        nvt/sllod       
nvt/sphere      oneway          pafi            pair            phonon          
pimd/langevin   pimd            pimd/nvt        planeforce      plumed          
pour            press/berendsen print           propel/self     property/atom   
python/invoke   python          python/move     qeq/comb        qeq/reaxff      
qeq/reax        rattle          reaxff/bonds    reax/c/bonds    reaxff/species  
reax/c/species  recenter        restrain        rhok            rigid           
rigid/nph       rigid/nph/small rigid/npt       rigid/npt/small rigid/nve       
rigid/nve/small rigid/nvt       rigid/nvt/small rigid/small     setforce        
sgcmc           shake           smd             spring          spring/chunk    
spring/rg       spring/self     srd             store/force     store/state     
temp/berendsen  temp/csld       temp/csvr       temp/rescale    tfmc            
thermal/conductivity            ti/spring       tmd             ttm             
ttm/grid        ttm/mod         tune/kspace     vector          viscosity       
viscous         viscous/sphere  wall/body/polygon               
wall/body/polyhedron            wall/colloid    wall/ees        wall/gran       
wall/gran/region                wall/harmonic   wall/lj1043     wall/lj126      
wall/lj93       wall/morse      wall/piston     wall/reflect    
wall/reflect/stochastic         wall/region     wall/region/ees wall/srd        
wall/table      widom           

* Compute styles:

ackland/atom    adf             aggregate/atom  angle           angle/local     
angmom/chunk    ave/sphere/atom basal/atom      body/local      bond            
bond/local      born/matrix     centro/atom     centroid/stress/atom            
chunk/atom      chunk/spread/atom               cluster/atom    cna/atom        
cnp/atom        com             com/chunk       contact/atom    coord/atom      
count/type      damage/atom     mesont          dihedral        dihedral/local  
dilatation/atom dipole          dipole/chunk    dipole/tip4p    
dipole/tip4p/chunk              displace/atom   efield/wolf/atom                
entropy/atom    erotate/asphere erotate/rigid   erotate/sphere  
erotate/sphere/atom             event/displace  fabric          fragment/atom   
global/atom     group/group     gyration        gyration/chunk  gyration/shape  
gyration/shape/chunk            heat/flux       hexorder/atom   hma             
improper        improper/local  inertia/chunk   ke              ke/atom         
ke/rigid        mliap           momentum        msd             msd/chunk       
msd/nongauss    omega/chunk     orientorder/atom                pair            
pair/local      pe              pe/atom         plasticity/atom pressure        
pressure/alchemy                property/atom   property/chunk  property/grid   
property/local  rdf             reduce          reduce/chunk    reduce/region   
rigid/local     slice           sna/atom        sna/grid        sna/grid/local  
snad/atom       snap            snav/atom       stress/atom     stress/cartesian                
stress/cylinder pressure/cylinder               stress/mop      
stress/mop/profile              stress/spherical                temp            
temp/asphere    temp/body       temp/chunk      temp/com        temp/cs         
temp/deform     temp/partial    temp/profile    temp/ramp       temp/region     
temp/rotate     temp/sphere     ti              torque/chunk    vacf            
vcm/chunk       viscosity/cos   voronoi/atom    

* Region styles:

block           cone            cylinder        ellipsoid       intersect       
plane           prism           sphere          union           

* Dump styles:

atom            atom/mpiio      cfg             cfg/mpiio       custom          
custom/mpiio    dcd             grid            grid/vtk        h5md            
image           local           movie           netcdf          xtc             
xyz             xyz/mpiio       yaml            

* Command styles

angle_write     balance         change_box      create_atoms    create_bonds    
create_box      delete_atoms    delete_bonds    box             kim_init        
kim_interactions                kim_param       kim_property    kim_query       
reset_ids       reset_atom_ids  reset_mol_ids   message         server          
dihedral_write  displace_atoms  dynamical_matrix                group2ndx       
hyper           info            kim             minimize        ndx2group       
neb             plugin          prd             read_data       read_dump       
read_restart    replicate       rerun           run             set             
tad             temper          temper/grem     temper/npt      third_order     
velocity        write_coeff     write_data      write_dump      write_restart

**Reply by njzjz** [reply]
Plugins cannot be listed by `lmp -h`.  See https://github.com/deepmodeling/deepmd-kit/issues/1809#issuecomment-1773111722

**Comment by Gemstrom** [comment]
> The plugin is only built for the MPI version. The serial one does not build the shared library so no plugin can be used.

Thank you so much! By the way, I wonder if lammps from conga-forge is incompatible with deepmd package?

**Reply by njzjz** [reply]
LAMMPS from conda-forge is compatible with deepmd-kit from conda-forge.

**Comment by Barrybee01** [comment]
I know this is a two year old post, but I also had this problem when trying to build the deepmd-kit plugin into LAMMPS. The source of this problem, for me at least, was missing a line in your input script. During the installation of deepmd and LAMMPS, you should end up with a /deepmd/lib/ folder in the directory you are building this code in, and in this folder are a set of plugin files. When using deepmd with LAMMPS, you need to have the following line before the "pair style deepmd" command:

plugin load libdeepmd_lmp.so

If this does not work, then the only other thing I can think of is a path issue during the installation. In my case, I am running LAMMPS simulations via a super computer, so there is a bash script I need to use to actually start the job. In this bash script, I have to load certain modules, activate my virtual environment, and set the path to the deepmd-kit source code. Here are the specific lines that I am using for this, but they can be modified to use for any other set up I imagine:

> module load gcc/9.3.0 cuda/11.0 openmpi/4.0.3
> module load python/3.10.2 cudnn/8.0.3 mpi4py/3.0.3
> source $HOME/venvs/deepmd/bin/activate
> deepmd_root=$HOME/software/deepmd
> export PATH="$deepmd_root/bin:$PATH"

I hope this helps anyone else who may have this specific problem

**Comment by tpchuckles** [comment]
I also ran into this issue. I wasn't able to get it working when installing with [pip](https://docs.deepmodeling.com/projects/deepmd/en/r2/getting-started/install.html#install-python-interface-with-pip), but the [conda](https://docs.deepmodeling.com/projects/deepmd/en/r2/getting-started/install.html#install-with-conda) install worked (it took eons to resolve the environment, but eventually installed). I of course had to run "conda activate deepmd" before running lmp.

I also noticed deepmd did _not_ seem to show up in "lmp -h", and my lammps input file also needed "plugin load libdeepmd_lmp.so" before my pair_style line. 

(I think each of these little "gotchas" is in this thread in various comments, but I seemed to need all)
